src/os/bluestore/BlueStore.cc

```diff
@@ -15511,6 +15512,13 @@ void BlueStore::_shutdown_cache()
 int BlueStore::flush_cache(ostream *os)
 {
   dout(10) << __func__ << dendl;
+  {
```

Comment:
IMO this block is redundant for impacting deferred ops processing.
If such ops are present in kv_sync_thread queues they are to be processed without the additional push (AKA kv_sync_in_progress = true; kv_cond.notify_one();) since IIUC putting into deferred op queue does such a push by itself. If they aren't present - there is no need to push.

What possibly matters is the lock acquisition which results in waiting for completion of  all the running tasks within  kv_sync_thread. I.e. by acquiring the lock here one makes sure kv_sync_thread is in waiting and hence it has nothing to do and all the deferred ops have been flushed.
Which means there is need to push it once again...

Reply:
@ifed01 If `deferred_aggresive` is not set, then going into sleep is performed regardless of state of `deferred_done_queue` and `deferred_stable_queue`. Only new incoming op `kv_queue` can actually trigger kv_sync_thread to process.
I observed this behavior by seeing that "perf dump" `bluestore_pinned_onodes` is non-zero for unlimited time after randomwrite tests are finished.

Reply:
@aclamk - yep, you're right!

But don't we still need to wait for kv_sync_thread to complete the processing we've just triggered before proceeding? I.e. shouldn't we wait until kv_sync_in_progress is back to false?